modules/mono/csharp_script.cpp

```diff
@@ -1203,7 +1206,7 @@ bool CSharpLanguage::setup_csharp_script_binding(CSharpScriptBinding &r_script_b
 
 	// ¯\_(ツ)_/¯
 	const ClassDB::ClassInfo *classinfo = ClassDB::classes.getptr(type_name);
-	while (classinfo && !classinfo->exposed) {
+	while (classinfo && (!classinfo->exposed || classinfo->gdextension || ignored_types.has(classinfo->name))) {
```

Comment:
1. I would add a comment calling this out as a workaround / hack with a link to the issue / PR, to make it easier to find when eventually fixing this properly
2. Pretty sure that singletons have to be ignored as well, or the ignored types list is pointless (it would just try to instanciate static classes instead and still crash anyways)

Reply:
Scenario: A method returns an extension implemented physics server singleton instance.

When this method is called from C# a wrapper object has to be created.
This code determines what type the wrapper should be, the actual extension class is disqualified, next the server extension is explicitly ignored, next the physics server type itself is chosen.
The wrapper then needs to be instanciated: while the managed type that corresponds to the native type can be found, it cannot be instanciated because it is a static class and static classes cannot have instances, thus error and probably a crash.

Because `Engine.GetSingleton` exists, there has to be some existing handling of this case, but I am not sure how exactly it works and I am not at my PC to check.
Depending on how exactly this works, ignoring the extensions here is either pointless, because singleton instances are alredy taken care of elsewhere, or singleton types need to be ignored here as well so that wrappers of type `Object` or `RefCounted` are properly created.